module/Core/src/RedirectRule/Entity/RedirectCondition.php

```diff
@@ -64,6 +65,11 @@ public static function forGeolocationCityName(string $cityName): self
         return new self(RedirectConditionType::GEOLOCATION_CITY_NAME, $cityName);
     }
 
+    public static function forBeforeDate(string $date): self
+    {
+        return new self(RedirectConditionType::BEFORE_DATE, $date);
+    }
```

Comment:
I'm wondering if this should be typed as `Chronos`, and do the cast to string here.

```suggestion
    public static function forBeforeDate(Chronos $date): self
    {
        return new self(RedirectConditionType::BEFORE_DATE, $date->toAtomString());
    }
```

That would be more in line with methods like `forDevice`, where `DeviceType` is expected, not `string`, and the casting is done there appropriately, avoiding user errors.

That means the value needs to be converted from the input string to Chronos somewhere, but I think that's fine, as that can be done with [`normalizeDate`](https://github.com/shlinkio/shlink/blob/497429e6859f774671995cd3b80beba6a765b10d/module/Core/functions/functions.php#L99)